edge/pkg/servicebus/servicebus.go

```diff
@@ -88,72 +121,154 @@ func processMessage(msg *beehiveModel.Message) {
+			m := "error to marshal request msg content"
+			code := http.StatusBadRequest
+			if response, err := buildErrorResponse(msg.GetID(), m, code); err == nil {
+				beehiveContext.SendToGroup(modules.HubGroup, response)
+			}
+			return
 		}
-		return
-	}
 
-	//send message with resource to the edge part
-	operation := httpRequest.Method
-	targetURL := "http://127.0.0.1:" + r[0] + r[1]
-	resp, err := uc.HTTPDo(operation, targetURL, httpRequest.Header, httpRequest.Body)
-	if err != nil {
-		m := "error to call service"
-		code := http.StatusNotFound
-		klog.Errorf(m, err)
-		if response, err := buildErrorResponse(msg.GetID(), m, code); err == nil {
-			beehiveContext.SendToGroup(modules.HubGroup, response)
+		var httpRequest commonType.HTTPRequest
+		if err := json.Unmarshal(content, &httpRequest); err != nil {
+			m := "error to parse http request"
+			code := http.StatusBadRequest
+			klog.Errorf(m, err)
+			if response, err := buildErrorResponse(msg.GetID(), m, code); err == nil {
+				beehiveContext.SendToGroup(modules.HubGroup, response)
+			}
+			return
 		}
-		return
-	}
-	defer resp.Body.Close()
-	resp.Body = http.MaxBytesReader(nil, resp.Body, maxBodySize)
-	resBody, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		if err.Error() == "http: request body too large" {
-			err = fmt.Errorf("response body too large")
+
+		//send message with resource to the edge part
+		operation := httpRequest.Method
+		targetURL := "http://127.0.0.1:" + r[0] + r[1]
+		resp, err := uc.HTTPDo(operation, targetURL, httpRequest.Header, httpRequest.Body)
+		if err != nil {
+			m := "error to call service"
+			code := http.StatusNotFound
+			klog.Errorf(m, err)
+			if response, err := buildErrorResponse(msg.GetID(), m, code); err == nil {
+				beehiveContext.SendToGroup(modules.HubGroup, response)
+			}
+			return
 		}
-		m := "error to receive response, err: " + err.Error()
-		code := http.StatusInternalServerError
-		klog.Errorf(m, err)
-		if response, err := buildErrorResponse(msg.GetID(), m, code); err == nil {
-			beehiveContext.SendToGroup(modules.HubGroup, response)
+		defer resp.Body.Close()
+		resp.Body = http.MaxBytesReader(nil, resp.Body, maxBodySize)
+		resBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			if err.Error() == "http: request body too large" {
+				err = fmt.Errorf("response body too large")
+			}
+			m := "error to receive response, err: " + err.Error()
+			code := http.StatusInternalServerError
+			klog.Errorf(m, err)
+			if response, err := buildErrorResponse(msg.GetID(), m, code); err == nil {
+				beehiveContext.SendToGroup(modules.HubGroup, response)
+			}
+			return
 		}
-		return
+
+		response := commonType.HTTPResponse{Header: resp.Header, StatusCode: resp.StatusCode, Body: resBody}
+		responseMsg := beehiveModel.NewMessage(msg.GetID())
+		responseMsg.Content = response
+		responseMsg.SetRoute("servicebus", modules.UserGroup).SetResourceOperation("", beehiveModel.UploadOperation)
```

Comment:
beehiveModel.NewMessage(msg.GetID()).FillBody(response).SetRoute(.........)

Reply:
there are three string literal "servicebus" in this file, and can be replaced with definitions `modules.ServiceBusModuleName`

Reply:
okay it had been fixed :)